post about first Karakun training (#101)
karakun/karakun.github.io: _posts/2018-11-08-react-training-announced.md

---
layout: post
title:  'Karakun React training in 2019'
author: hendrik
featuredImage: training-1

> excerpt: 'TODO'

Reviewer: Todo

Reviewer: ```suggestion
excerpt: 'Beginning next year we will offer trainings to different topics in different locations. We're happy to announce that our first training is already able for booking. Simon will host a training about React in Munich. Hurry up and book your seat soon!'
```

categories: [JavaScript, React]
header:
  image: post
---

> 2019 beginnen wir eine Reihe an Trainings anzubieten.

Reviewer: ```suggestion
Beginning next year we will offer trainings to different topics in different locations. 
```

> Hier werden wir Dortmund, München & Basel als Orte haben

Reviewer: ```suggestion
You'll get trained by us in Basel (CH), Munich (D) and Dortmund (D).
```

> Erste Training (in München) bereits buchbar: https://eppleton.de/kurse/react-training-mit-simon-skozczylas_31.html

Reviewer: ```suggestion
We're happy to announce that our first training is already able for booking: 
[React Training](https://eppleton.de/kurse/react-training-mit-simon-skozczylas_31.html) with [Simon]({{ "/people/simon.html" | absolute_url }}) in Munich.
```

![React training]({{ "/assets/images/illustrations/karakun_react.png" | absolute_url }})

## Overview of the React training

> Die Web-Entwicklung gewinnt in der Software-Entwicklung immer mehr an Bedeutung. Um auch komplexere Projekte erfolgreich umsetzen zu können kommt in vielen Projekten die React Bibliothek zum Einsatz. React verspricht performante Anwendungen und gleichzeitig einen einfachen, klar strukturierten Aufbau. In diesem Training lernen Sie die Grundprinzipien von React.

Reviewer: ```suggestion
Web development is gaining more and more importance in software development. In order to successfully implement more complex projects, the React library is used in many projects. React promises high-performance applications and at the same time a simple, clearly defined structure. In this training you will learn the basic principles of React.
```

> Praktische Übungen ergänzen dabei die im Kurs vermittelte Theorie. Das gelernte Wissen wenden die Teilnehmer gleich in konkreten Beispielen aus dem Entwickler-Alltag an. Die neuen Skills können so direkt auf die eigene Entwicklungsarbeit im Projekt übertragen werden.

Reviewer: ```suggestion
Practical exercises support the theory taught in the course. The participants apply the skills they have learnt in concrete examples from their everyday work as developers. The new skills can thus be transferred directly to their own development work in the project.
```

> ## Die wichtigsten Themen im Überblick:

Reviewer: ```suggestion
## Outline with the most important Topics
```

> - Wieso ist es eine Bibliothek und kein Framework?

Reviewer: ```suggestion
* Why is React a library and not a framework?
```

> - Welche Konzepte stecken hinter React?

Reviewer: ```suggestion
* What are the concepts behind React?
```

> - Was sind die Props?

Reviewer: ```suggestion
* What are these "props"?
```

> - Was ist der State?

Reviewer: ```suggestion
* What are these "state"?
```

> - JavaScript ES6 Grundlagen

Reviewer: ```suggestion
* JavaScript ES6 basics
```

> - JavaScript syntax extension (JSX)

Reviewer: ```suggestion
* JavaScript syntax extension (JSX)
```

> - Lebenszyklus der Komponenten (Mount, Update, Error)

Reviewer: ```suggestion
* Components Life cycle
```

> - Build-Prozess

Reviewer: ```suggestion
* Build-Process
```

> - Der React Renderer (ReactDOM, ReactDOM Server, ReactVR, ...)

Reviewer: ```suggestion
* The React Renderer (ReactDOM, ReactDOM Server, ReactVR, ...)
```

> - State-Management (Context API, Redux, MobX)

Reviewer: ```suggestion
* State-Management (Context API, Redux, MobX)
```

> - Datenabrufen vom Server

Reviewer: ```suggestion
* Client-Server communication
```

> - Router

Reviewer: ```suggestion
* Router
```

> - Testen von React Komponenten

Reviewer: ```suggestion
* React components testing
```

> - PropTypes, Flow, TypeScript

Reviewer: ```suggestion
* PropTypes, Flow, TypeScript
```

---
layout: post
title:  'Karakun React training in 2019'

> author: hendrik

Author: ```suggestion
author: timo
```

Author: Du brauchst ein Foto!

featuredImage: training-1

> excerpt: 'Beginning next year we will offer trainings to different topics in different locations. We're happy to announce that our first training is already able for booking. Simon will host a training about React in Munich. Hurry up and book your seat soon!'

Author: ```suggestion
excerpt: 'Beginning next year we will offer trainings to different topics in different locations. We're happy to announce that our first training is already able for booking. Our JavaScript expert Simon Skoczylas will host a training about React in Munich. Hurry up and book your seat soon!'
```

categories: [JavaScript, React]
header:
  image: post
---

Beginning next year we will offer trainings to different topics in different locations. 
You'll get trained by us in Basel (CH), Munich (D) and Dortmund (D).
We're happy to announce that our first training is already able for booking: 
[React Training](https://eppleton.de/kurse/react-training-mit-simon-skozczylas_31.html) with [Simon]({{ "/people/simon.html" | absolute_url }}) in Munich.

![React training]({{ "/assets/images/illustrations/karakun_react.png" | absolute_url }})

## Overview of the React training
Web development is gaining more and more importance in software development. In order to successfully implement more complex projects, the React library is used in many projects. React promises high-performance applications and at the same time a simple, clearly defined structure. In this training you will learn the basic principles of React.

Practical exercises support the theory taught in the course. The participants apply the skills they have learnt in concrete examples from their everyday work as developers. The new skills can thus be transferred directly to their own development work in the project.

> ## Outline with the most important Topics

Author: ```suggestion
### Outline with the most important Topics
```

* Why is React a library and not a framework?
* What are the concepts behind React?
* What are these "props"?
* What are these "state"?
* JavaScript ES6 basics
* JavaScript syntax extension (JSX)

> * Components Life cycle

Author: ```suggestion
* Components lifecycle
```

* Build-Process

> * The React Renderer (ReactDOM, ReactDOM Server, ReactVR, ...)

Author: ```suggestion
* The React renderer (ReactDOM, ReactDOM Server, ReactVR, ...)
```
